src/psyclone/psyir/transformations/matmul2code_trans.py

```diff
@@ -151,11 +198,14 @@ def validate(self, node):
             # There should be one index per dimension. This is enforced
             # by the array create method so is not tested here.
 
-            # The first two indices should be ranges. This
-            # transformation is currently limited to Ranges which
-            # specify the full extent of the dimension.
+            # The first two indices should be ranges. This is a
+            # limitation of this transformation, not the PSyIR, as it
+            # would be valid to have any two indices being
+            # ranges. Further, this transformation is currently
+            # limited to Ranges which specify the full extent of the
+            # dimension.
             if not (matrix.is_full_range(0) and matrix.is_full_range(1)):
-                raise GenerationError(
+                raise TransformationError(
```

Comment:
Thanks for making the comment more explicit. Perhaps this error and the next one should be `NotImplementedError` to more accurately reflect what's going on?

Reply:
Changed. However, the tests are a mix of general correctness (e.g. there should be 2 dimensions with ranges) and constraints (e.g. these can be any two ranges but are currently limited to being the first two). This logic will need to be modified when ranges in any dimension is supported.